app/actions/handlers.py

```diff
@@ -14,70 +14,42 @@
 state_manager = IntegrationStateManager()
 
 
-async def filter_and_transform(vehicles, transmissions, integration_id, action_id):
+async def validate_gmt_offset(serial_num, gmt_offset, integration_id, action_id):
+    # Check for invalid offsets
+    if gmt_offset is None:
+        return 0
+    elif abs(gmt_offset) <= 24:
+        return gmt_offset
+    else:
+        message = f"GMT offset invalid for device '{serial_num}' value '{gmt_offset}'"
+        logger.error(
+            message,
+            extra={
+                'needs_attention': True,
+                'integration_id': integration_id,
+                'action_id': action_id
+            }
+        )
+        return None
+
+
+async def filter_and_transform(serial_num, vehicles, gmt_offsets_per_device, integration_id, action_id):
     transformed_data = []
     main_data = ["ats_serial_num", "date_year_and_julian", "latitude", "longitude"]
 
-    vehicles_to_skip = []  # FOR NOW, THESE VEHICLES WILL BE SKIPPED BECAUSE NO GMT OFFSET SET
-
-    # Get GMT offset per serial num
-    serial_nums = set([v.ats_serial_num for v in vehicles])
-    gmt_offsets = {}
-    for serial_num in serial_nums:
-        try:
-            gmt_offset = next(iter(set([t.gmt_offset for t in transmissions if t.collar_serial_num == serial_num])))
-            # check for invalid offsets
-            if gmt_offset is None:
-                gmt_offsets[serial_num] = 0
-            elif abs(gmt_offset) <= 24:
-                gmt_offsets[serial_num] = gmt_offset
-            else:
-                vehicles_to_skip.append(serial_num)
-                message = f"GMT offset invalid for device '{serial_num}' value '{gmt_offset}'"
-                logger.warning(
-                    message,
-                    extra={
-                        'needs_attention': True,
-                        'integration_id': integration_id,
-                        'action_id': action_id
-                    }
-                )
-        except StopIteration:
-            # If no offset found, a 0 offset will be set (new collar)
-            gmt_offsets[serial_num] = 0
+    # Get GMT offset
+    offset = await validate_gmt_offset(serial_num, gmt_offsets_per_device.get(serial_num), integration_id, action_id)
```

Comment:
Instead of doing the validate logic on each vehicle, just precalculate the offset values for all the vehicles you know about.
Let that pre-calculated dictionary include all the serial_numbers that you have a valid offset for. Then here, do a `gmt_offsets.get(serial_num, 0)`.
This will coerce both missing serial numbers and those invalid values to be zero.

Reply:
In otherwords, let's send everything to ER, even if the timestamp is wrong (in this case wrong by having bad data coming from ATS).

If we are able to , then let's add the original gmt offset value to the `Observation.additional` data before sending to Sensors API.